lib/archethic/mining/validation_context.ex

```diff
@@ -1328,14 +1327,18 @@ defmodule Archethic.Mining.ValidationContext do
 
   @doc """
   Return the list of nodes which confirmed the transaction replication
+
+  We use the authorized_and_available_nodes as a common reference between the nodes
   """
   @spec get_confirmed_replication_nodes(t()) :: list(Node.t())
   def get_confirmed_replication_nodes(%__MODULE__{
-        chain_storage_nodes: chain_storage_nodes,
+        validation_time: validation_time,
         storage_nodes_confirmations: storage_nodes_confirmations
       }) do
+    nodes = P2P.authorized_and_available_nodes(validation_time)
```

Comment:
I think you should use ` Enum.sort_by(& &1.first_public_key)` to maintain the same order as in the `ReplicationAttestation.get_node_index`, which was used to determine the index in the confirmations.